includes/object-cache.php

```diff
@@ -1160,13 +1160,77 @@ public function add_multiple( array $data, $group = 'default', $expire = 0 ) {
             return array_combine( array_keys( $data ), array_fill( 0, count( $data ), false ) );
         }
 
-        $values = [];
+        $results = [];
+
+        if (! $this->is_ignored_group($group) && $this->redis_status() ) {
+            $orig_exp = $expire;
+            $expire = $this->validate_expiration( $expire );
+            $tx = $this->redis->pipeline();
+            $keys = array_keys($data);
+
+            foreach ( $data as $key => $value ) {
+                /**
+                 * Filters the cache expiration time
+                 *
+                 * @since 1.4.2
+                 * @param int    $expiration The time in seconds the entry expires. 0 for no expiry.
+                 * @param string $key        The cache key.
+                 * @param string $group      The cache group.
+                 * @param mixed  $orig_exp   The original expiration value before validation.
+                 */
+                $expire = apply_filters( 'redis_cache_expiration', $expire, $key, $group, $orig_exp );
+                $start_time = microtime( true );
+
+                $derived_key = $this->fast_build_key( $key, $group );
+                $results[ $key ] = $derived_key || ! isset( $this->cache[ $derived_key ] );
+                $args = [ $derived_key, $this->maybe_serialize( $value ) ];
+
+                if ( $this->redis instanceof Predis\Client ) {
+                    $args[] = 'nx';
+                    $args[] = 'ex';
+                    $args[] = $expire;
+                } else {
+                    $args[] = [
+                        'nx',
+                        'ex' => $expire,
+                    ];
+                }
+
+                $tx->set( ...$args );
+
+                $execute_time = microtime( true ) - $start_time;
+
+                if ( $this->trace_enabled ) {
+                    $this->trace_command( 'set', $group, [
+                        $key => [
+                            'value' => $value,
+                            'status' => self::TRACE_FLAG_WRITE,
+                        ],
+                    ], microtime( true ) - $start_time );
+                }
+
+                $this->cache_calls++;
+                $this->cache_time += $execute_time;
+            }
+            try {
+
+                $method = ( $this->redis instanceof Predis\Client ) ? 'execute' : 'exec';
+
+                $results = array_map( function ( $response ) {
+                    return (bool) $this->parse_redis_response( $response );
+                }, $tx->{$method}() );
+
+                $results = array_combine( $keys, $results );
+
+            } catch ( Exception $exception ) {
+                $this->handle_exception( $exception );
+
+                $results[$key] = false;
+            }
 
-        foreach ( $data as $key => $value ) {
-            $values[ $key ] = $this->add( $key, $value, $group, $expire );
         }
 
-        return $values;
+        return $results;
```

Comment:
I think we should do the same wrapper:

https://github.com/rhubarbgroup/redis-cache/blob/56ff6116eb00cf4dc54e5bff1fcbce2d238dcd65/includes/object-cache.php#L1370-L1382

Reply:
you mean using 
```
     if ( $this->redis_status() && method_exists( $this->redis, 'pipeline' ) ) { 
         return $this->delete_multiple_at_once( $keys, $group ); 
     } 
```

or using `values` instead of `results`

Reply:
Just two methods, one with the fallback of pipeline() isn't supported.

Reply:
Let's use the same approach with a dedicated `*_multiple_at_once()` method to make this more readable.

Reply:
@tillkruss done, can you check.